hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -546,6 +646,258 @@ final class VCFLine(val line: String, arrayElementsRequired: Boolean) {
+        parseIntInfoArrayElement()
+      }
+
+      rvb.startArray(abi.length)
+      var i = 0
+      while (i < abi.length) {
+        if (abi.isMissing(i))
+          rvb.setMissing()
+        else
+          rvb.addInt(abi(i))
+        i += 1
+      }
+      rvb.endArray()
+      abi.clear()
+    }
+  }
+
+  def parseAddInfoArrayString(rvb: RegionValueBuilder) {
+    if (!infoFieldMissing()) {
+      rvb.setPresent()
+      assert(abs.length == 0)
+      parseStringInfoArrayElement()
+      while (!endInfoField()) {
+        pos += 1  // comma
+        parseStringInfoArrayElement()
+      }
+
+      rvb.startArray(abs.length)
+      var i = 0
+      while (i < abs.length) {
+        if (abs.isMissing(i))
+          rvb.setMissing()
+        else
+          rvb.addString(abs(i))
+        i += 1
+      }
+      rvb.endArray()
+      abs.clear()
+    }
+  }
+
+  def parseAddInfoArrayDouble(rvb: RegionValueBuilder) {
+    if (!infoFieldMissing()) {
+      rvb.setPresent()
+      assert(abd.length == 0)
+      parseDoubleInfoArrayElement()
+      while (!endInfoField()) {
+        pos += 1  // comma
+        parseDoubleInfoArrayElement()
+      }
+
+      rvb.startArray(abd.length)
+      var i = 0
+      while (i < abd.length) {
+        if (abd.isMissing(i))
+          rvb.setMissing()
+        else
+          rvb.addDouble(abd(i))
+        i += 1
+      }
+      rvb.endArray()
+      abd.clear()
+    }
+  }
+
+  def parseAddInfoField(rvb: RegionValueBuilder, typ: Type) {
+    val c = line(pos)
+    if (c != ';' && c != '\t') {
+      if (c != '=')
+        parseError(s"invalid INFO key/value expression found '${line(pos)}' instead of '='")
+      pos += 1 // equals
+      typ match {
+        case TInt32(_) => parseAddInfoInt(rvb)
+        case TString(_) => parseAddInfoString(rvb)
+        case TFloat32(_) => parseAddInfoFloat(rvb)
+        case TFloat64(_) => parseAddInfoDouble(rvb)
+        case TArray(TInt32(_), _) => parseAddInfoArrayInt(rvb)
+        case TArray(TFloat64(_), _) => parseAddInfoArrayDouble(rvb)
+        case TArray(TString(_), _) => parseAddInfoArrayString(rvb)
+      }
+    }
+  }
+
+  def parseAddInfo(rvb: RegionValueBuilder, infoType: TStruct, infoFlagFieldNames: Set[String]) {
+    rvb.startStruct()
+    infoType.fields.foreach { f =>
+      if (infoFlagFieldNames.contains(f.name))
+        rvb.addBoolean(false)
+      else
+        rvb.setMissing()
+    }
+
+    // handle first key, which may be '.' for missing info
+    var key = parseInfoKey()
+    if (key == ".") {
+      if (endField()) {
+        rvb.endStruct()
+        return
+      } else
+        parseError(s"invalid INFO key $key")
+    }
+
+    if (infoType.hasField(key)) {
+      rvb.setFieldIndex(infoType.fieldIdx(key))
+      if (infoFlagFieldNames.contains(key))
+        rvb.addBoolean(true)
+      else
+        parseAddInfoField(rvb, infoType.fieldType(key))
+    }
+    skipInfoField()
+
+    while (!endField()) {
+      nextInfoField()
```

Comment:
Can you get rid of the code above and just keep this while loop?

Reply:
Sorry I meant this:
```
    if (infoType.hasField(key)) {
       rvb.setFieldIndex(infoType.fieldIdx(key))
       if (infoFlagFieldNames.contains(key))
         rvb.addBoolean(true)
       else
         parseAddInfoField(rvb, infoType.fieldType(key))
     }
     skipInfoField()
```

Reply:
this is the alternative, thoughts?
```scala
    var key = parseInfoKey()
    if (key == ".") {
      if (endField()) {
        rvb.endStruct()
        return
      } else
        parseError(s"invalid INFO key $key")
    }

    while (!endField()) {
      if (key == ".") {
        parseError(s"invalid INFO key $key")
      }
      if (infoType.hasField(key)) {
        rvb.setFieldIndex(infoType.fieldIdx(key))
        if (infoFlagFieldNames.contains(key))
          rvb.addBoolean(true)
        else
          parseAddInfoField(rvb, infoType.fieldType(key))
      }
      skipInfoField()
      if (!endField()) {
        nextInfoField()
        key = parseInfoKey()
      }
    }
```

Reply:
This seems better. So we just skip a field silently if it's not in the `infoType`?